ui/components/app/modals/visit-support-data-consent-modal/visit-support-data-consent-modal.tsx

```diff
@@ -0,0 +1,156 @@
+import React, { useCallback, useContext } from 'react';
+import { useSelector } from 'react-redux';
+import { useI18nContext } from '../../../../hooks/useI18nContext';
+import { selectSessionData } from '../../../../selectors/identity/authentication';
+import { getMetaMetricsId } from '../../../../selectors/selectors';
+import { openWindow } from '../../../../helpers/utils/window';
+import {
+  Modal,
+  ModalOverlay,
+  ModalContent,
+  ModalHeader,
+  Box,
+  ModalFooter,
+  ButtonPrimary,
+  ButtonPrimarySize,
+  ModalBody,
+  Text,
+  ButtonSecondary,
+  ButtonSecondarySize,
+} from '../../../component-library';
+import {
+  Display,
+  TextVariant,
+  BlockSize,
+} from '../../../../helpers/constants/design-system';
+import {
+  MetaMetricsContextProp,
+  MetaMetricsEventCategory,
+  MetaMetricsEventName,
+} from '../../../../../shared/constants/metametrics';
+import { MetaMetricsContext } from '../../../../contexts/metametrics';
+import { SUPPORT_LINK } from '../../../../../shared/lib/ui-utils';
+
+type VisitSupportDataConsentModalProps = {
+  onClose: () => void;
+  isOpen: boolean;
+};
+
+const VisitSupportDataConsentModal: React.FC<
+  VisitSupportDataConsentModalProps
+> = ({ isOpen, onClose }) => {
+  const version = process.env.METAMASK_VERSION;
+  const t = useI18nContext();
+  const trackEvent = useContext(MetaMetricsContext);
+  const sessionData = useSelector(selectSessionData);
+  const profileId = sessionData?.profile?.profileId;
+  const metaMetricsId = useSelector(getMetaMetricsId);
+
+  const handleClickContactSupportButton = useCallback(
+    (params: {
+      version: string;
+      profileId?: string;
+      metaMetricsId?: string;
+    }) => {
+      onClose();
+      let supportLinkWithUserId = SUPPORT_LINK;
+      const queryParams = new URLSearchParams();
+      queryParams.append('metamask_version', params.version);
+      if (params.profileId) {
+        queryParams.append('metamask_profile_id', params.profileId);
+      }
+      if (params.metaMetricsId) {
+        queryParams.append('metamask_metametrics_id', params.metaMetricsId);
+      }
```

Comment:
@itsyoboieltr Thanks for spotting the issue for undefined, they should be handled here now 🙏🏻